scripts/xtensa-build-all.sh

```diff
@@ -30,30 +39,56 @@ usage: xtensa-build.sh [options] platform(s)
        -u Force UP ARCH
        -d Enable debug build
        -c Interactive menuconfig
+       -o copies the file argument from src/arch/xtensa/configs/override/$arg.config
+	  to the build directory after invoking CMake and before Make.
        -k Use private key
        -v Verbose Makefile log
        -j n Set number of make build jobs. Jobs=#cores when no flag. \
 Infinte when not specified.
+	-m path to MEU tool. Switches signing step to use MEU instead of rimage.
+	-s path to openssl tool. Requires -m and required by -m.
        Supported platforms ${SUPPORTED_PLATFORMS[*]}
 EOF
 }
 
 # parse the args
-while getopts "rudj:ckva" OPTION; do
+while getopts "rudj:ck:vao:m:s:" OPTION; do
         case "$OPTION" in
 		r) BUILD_ROM=yes ;;
 		u) BUILD_FORCE_UP=yes ;;
 		d) BUILD_DEBUG=yes ;;
 		j) BUILD_JOBS=$OPTARG ;;
 		c) MAKE_MENUCONFIG=yes ;;
-		k) USE_PRIVATE_KEY=yes ;;
+		o) OVERRIDE_CONFIG=$OPTARG ;;
+		k) PRIVATE_KEY=$OPTARG ;;
 		v) BUILD_VERBOSE='VERBOSE=1' ;;
 		a) PLATFORMS=("${SUPPORTED_PLATFORMS[@]}") ;;
+		m) MEU_TOOL_PATH=$OPTARG ;;
+		s) MEU_OPEN_SSL_PATH=$OPTARG ;;
 		*) print_usage; exit 1 ;;
         esac
 done
 shift $((OPTIND-1))
 
+#default signing tool
+SIGNING_TOOL=RIMAGE
+
+if [ -n "${OVERRIDE_CONFIG}" ]
+then
+	OVERRIDE_CONFIG="src/arch/xtensa/configs/override/$OVERRIDE_CONFIG.config"
+	[ -f "${OVERRIDE_CONFIG}" ] || die 'Invalid override config file %s\n' "${OVERRIDE_CONFIG}"
+fi
+
+if [ -n "${MEU_TOOL_PATH}" ]
+then
+	[ -d "${MEU_TOOL_PATH}" ] || die 'Invalid MEU TOOL PATH %s\n' "${MEU_TOOL_PATH}"
+	[ -f "${MEU_OPEN_SSL_PATH}" ] ||  die 'Invalid OPENSSL PATH %s\n' "${MEU_OPEN_SSL_PATH}"
```

Comment:
Is `-DMEU_OPENSSL` really required for using MEU? I just looked at `src/arch/xtensa/CMakeLists.txt` and there is a default value there (`/usr/bin/openssl`)

Reply:
@marc-hb Some meu has issue with OpenSSL 1.1. This will bring us a way to override the setting to use OpenSSL 1.0
But I think this may be not needed, as the issue is already be fixed.

Reply:
@xiulipan the issue still exists AFAIK. I had to use OpenSSL1.0 for this to work on my side

Reply:
> This will bring us a way to override the setting to use OpenSSL 1.0

Sorry I wasn't clear, I meant: do we need to FORCE the user to provide this value? It think it should be optional, we should just pass-through the either empty or non-empty value without looking at it, same as pass-through commit f7f9d5f42e5896fb08 did.

In this script much less code and no logic, just 30 characters:
```shell
cmake -DMEU_OPENSSL="${MEU_OPENSSL}" ....
```
If needed, override CMake default with: `MEU_OPENSSL=~/openssl1.0/ ./scripts/xtensa-build-all.sh ....` , if not needed pass nothing and use the existing CMake default that I just discovered.

@ranj063 ?

Reply:
@marc-hb so you want me to make the -s optional?

Reply:
No, I'm suggesting the `-s` goes away replaced with an environment variable $MEU_OPENSSL. See example above. Unlike the much more complex and more important situation with the private keys, the MEU_OPENSSL location is just a configuration detail that doesn't require any logic, it can just be passed-through this script.

Reply:
done! fixed this along with a new patch to fix cmake as well